envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -55,6 +63,18 @@ internal class EnvoySnapshotFactory(
         )
     }
 
+    private fun toClusterConfiguration(instances: List<ServiceInstances>, serviceName: String): ClusterConfiguration {
+        val allInstances = instances.flatMap {
+            it.instances
+        }
+
+        val allInstancesHaveEnvoyTag = allInstances.isNotEmpty() && allInstances.all {
+            it.tags.contains("envoy")
+        }
+
+        return ClusterConfiguration(serviceName, allInstancesHaveEnvoyTag)
```

Comment:
Why do we enable http2 only if all instances have envoy tag? Unclear to me.

Reply:
Http2 support is on a cluster level so if someone decides to deploy a service in dc1 with envoy and in dc2 without envoy then we can't set http2 because we do not know if the server in dc2 supports it.